code/modules/mob/living/carbon/human/species.dm

```diff
@@ -95,6 +102,14 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	//The component to add when swimming
 	var/swimming_component = /datum/component/swimming
 
+	//K-Limbs. If a species doesn't have their own limb types. Do not override this, use the K-Limbs overrides at the top of the species datum.
+	var/obj/item/bodypart/species_chest = /obj/item/bodypart/chest
+	var/obj/item/bodypart/species_head = /obj/item/bodypart/head
+	var/obj/item/bodypart/species_l_arm = /obj/item/bodypart/l_arm
+	var/obj/item/bodypart/species_r_arm = /obj/item/bodypart/r_arm
+	var/obj/item/bodypart/species_r_leg = /obj/item/bodypart/r_leg
+	var/obj/item/bodypart/species_l_leg = /obj/item/bodypart/l_leg
```

Comment:
Do these variables get replaced somewhere with `species_chest  = new species_chest `? If not then they are a typepath and not an /obj/item/bodypart and shouldn't be typed as such. The only bit that is kinda dumb in byond is that you need to type a typepath as an object if you are using initial on uninstantiated types.
```
var/obj/item/bodypart/part = /obj/item/bodypart/chest
part = new part() //in initialize somewhere so whenever you check part its an object and not a type
```
In the case of using initial I would use it as a local variable to avoid confusion
```
var/obj/item/bodypart/part = type_path
if(initial(part.variable) == 4)
...
```

Reply:
Also I agree with it being an assoc list, although I wouldn't put typepaths in that list.
var/list/parts = list()

Best way I can see doing it is having a var on species that defines leg_typepath, arm_typepath etc... and then having the part list that holds a dictionary of the instantiated bodyparts
```
//Somewhere in initialization or new
parts[BODYPART_CHEST] = new species_chest(src)
```

Reply:
see species/proc/replace_body() in species.dm. I’m on mobile I will give more thorough answers when I get home.

Reply:
> Do these variables get replaced somewhere with species_chest = new species_chest ?

Yes. species/proc/replace_body() does a similar thing.

![image](https://user-images.githubusercontent.com/75460809/142712188-3d65f6c0-f33f-491e-9ffc-02bf05a611eb.png)
already happens in the above proc


![image](https://user-images.githubusercontent.com/75460809/142712214-f23e1e3b-facb-4f6a-a3ea-eeffbc4958d1.png)

So the exact same thing but also with a list holding every bodypart to make it slightly easier to reference? Is that really worth the added memory cost of a list?

Reply:
Yes. Memory hasn't really been an issue since we got large address aware